### ISSUE TITLE
Problem with JSON function encoding

### ISSUE THREAD

**Issue by fabriciogpp** [huandu/go-sqlbuilder#171]
I'm trying to upgrade from `v1.27.3` to `v1.30.0` and it is giving me this issue.

I have this statement composing my query:

```go
sb.Equal("data->'$.some.field'", 123)
```

And it encoded it like this:

```sql
SELECT COUNT(*) FROM table WHERE id = ? AND deleted_at IS NULL AND data->'$$.some.field' = ?
```

Which produced this error:

```Error 3143 (42000): Invalid JSON path expression. The error is around character position 1.```

Note the double `$$` in the query.

I manually replaced it with `strings.ReplaceAll(qry, "$$", "$")`  and the query worked again.

Is this a bug or I'm missing something?

---

Edit: with version `v1.29.0` works fine. It is a problem with `v1.30.0`.

---

(my driver is `github.com/go-sql-driver/mysql v1.8.1`)
(full builder code would be something like this:)
```go
sb := sqlbuilder.Select("COUNT(*)").From("table")
where := []string{ sb.Equal("data->'$.some.field'", 123), /* more... */}
qry, args := sb.Where(where...).Build()
```

**Comment by huandu** [comment]
It's a bug. Thanks for reporting this to me.

**Comment by huandu** [comment]
Fixed in tag v1.30.1.
